tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_InitialDataTci.cpp

```diff
@@ -0,0 +1,143 @@
+      1.0e-20, 1.0e-40, 1.1e-12, 1.0e-12, std::optional<double>{1.0e-2}};
+
+  {
+    INFO("TCI is happy");
+    const InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(),
+                                        1.0};
+    CHECK_FALSE(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, delta0, epsilon, exponent, dg_mesh,
+        tci_options));
+  }
+
+  {
+    INFO("Two mesh RDMP fails");
+    const InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(),
+                                        2.0};
+    CHECK(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, delta0, epsilon, exponent, dg_mesh,
+        tci_options));
+  }
+
+  {
+    INFO("Persson TCI TildeD fails");
+    const InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(),
+                                        1.0};
+    get(get<grmhd::ValenciaDivClean::Tags::TildeD>(
+        dg_vars))[dg_mesh.number_of_grid_points() / 2] += 2.0e10;
+    CHECK(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, 1.0e100, epsilon, exponent, dg_mesh,
+        tci_options));
+    get(get<grmhd::ValenciaDivClean::Tags::TildeD>(
+        dg_vars))[dg_mesh.number_of_grid_points() / 2] = 1.0;
+  }
+
+  {
+    INFO("Persson TCI TildeB fails");
+    const InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(),
+                                        1.0};
+    for (size_t i = 0; i < 3; ++i) {
+      get<grmhd::ValenciaDivClean::Tags::TildeB<>>(dg_vars).get(
+          i)[dg_mesh.number_of_grid_points() / 2] += 2.0e10;
+    }
+    CHECK(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, 1.0e100, epsilon, exponent, dg_mesh,
+        tci_options));
+    for (size_t i = 0; i < 3; ++i) {
+      get<grmhd::ValenciaDivClean::Tags::TildeB<>>(dg_vars).get(
+          i)[dg_mesh.number_of_grid_points() / 2] = 1.0;
+    }
+  }
+
+  {
+    INFO("Persson TCI pressure fails");
+    const InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(),
+                                        1.0};
+    get(get<grmhd::ValenciaDivClean::Tags::TildeTau>(
+        dg_vars))[dg_mesh.number_of_grid_points() / 2] += 2.0e10;
+    CHECK(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, 1.0e100, epsilon, exponent, dg_mesh,
+        tci_options));
+    get(get<grmhd::ValenciaDivClean::Tags::TildeTau>(
+        dg_vars))[dg_mesh.number_of_grid_points() / 2] = 1.0;
+  }
+
+  {
+    INFO("Negative TildeD");
+    InactiveConsVars subcell_vars{subcell_mesh.number_of_grid_points(), 1.0};
+
+    get(get<grmhd::ValenciaDivClean::Tags::TildeD>(
+        dg_vars))[dg_mesh.number_of_grid_points() / 2] = -1.0e-20;
+    CHECK(grmhd::GhValenciaDivClean::subcell::DgInitialDataTci::apply(
+        dg_vars, subcell_vars, 1.0e100, epsilon, exponent, dg_mesh,
```

Comment:
We need to relax both Persson and RDMP TCI here (and a few tests after this line as well) to ensure that `false` comes from the negativity of `TildeD`, but I believe Persson TCI might trigger here if `epsilon` and `exponent` are not changed from the preset values. One way might be using a huge `epsilon` or a large negative value of `exponent`.